test_tracetools/src/mark_process.cpp

```diff
@@ -42,13 +47,19 @@ void mark_trace_test_process()
 {
 #ifndef TRACETOOLS_DISABLED
   // See tracetools_test.mark_process for more details
-  const std::string env_var{trace_test_id_env_var};
-  const auto test_id = rcpputils::get_env_var(env_var.c_str());
+  const std::string runtime_disable_env_var{tracetools_runtime_disable_env_var};
+  const auto runtime_disable = rcpputils::get_env_var(runtime_disable_env_var.c_str());
+  if (!runtime_disable.empty() && runtime_disable == "1") {
+    return;
+  }
```

Comment:
I'm not sure we should apply the mechanism here. This process marking thing is meant to be a test functionality, so it shouldn't necessarily be disabled if `TRACETOOLS_RUNTIME_DISABLE` is enabled.

Also, in that case, given the process marking functionality, the `test_runtime_disable` test in `test_ros2trace` doesn't actually test the runtime disabling mechanism, right? In practice, the test would still pass even if the nodes incorrectly generated trace events, simply because there would be no process with the given trace event markers. Therefore, there would be no trace events associated with this test as per `get_corresponding_trace_test_events()`.

Let me know if that makes sense.